src/bp-core/bp-core-catchuri.php

```diff
@@ -1545,3 +1545,77 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 
 	return $link;
 }
+
+/**
+ * Function will remove default wordpress and woocommerce api endpoint.
+ * Also, remove feeds url.
+ */
+add_action('init', 'bb_removes_endpoints_and_feeds');
+function bb_removes_endpoints_and_feeds() {
+	if ( is_user_logged_in() ) {
+		return;
+	}
+	
+	// IF Platform 'Private Website' settings enabled.
+	// THEN Restrict all RSS and REST APIs. (reason to restrict because the site is private).
+	if ( ! bp_enable_private_network() ) {
+		bb_restricate_rss_feed();
+	}
+	
+	// check settings with BuddyBoss APP and Platform both.
+	if ( bp_rest_enable_private_network() ) {
+		bb_restricate_rest_api();
+	}
+}
+
+/**
+ * Function will restricate RSS feed.
+ */
+function bb_restricate_rss_feed() {
+	// This will disable default feeds.
+	remove_action( 'do_feed_rdf', 'do_feed_rdf', 10, 1 );
+	remove_action( 'do_feed_rss', 'do_feed_rss', 10, 1 );
+	remove_action( 'do_feed_rss2', 'do_feed_rss2', 10, 1 );
+	remove_action( 'do_feed_atom', 'do_feed_atom', 10, 1 );
+	remove_action( 'wp_head', 'feed_links_extra', 3 );
+	remove_action( 'wp_head', 'feed_links', 2 );
+	// This will disable sitewide feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_sitewide_feed' );
+	// This will disable users perosnal activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_personal_feed' );
+	// This will disable users friends activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_friends_feed' );
+	// This will disable users group activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_my_groups_feed' );
+	// This will disable users mentions activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_mentions_feed' );
+	// This will disable users favorites activity feeds.
+	remove_action( 'bp_actions', 'bp_activity_action_favorites_feed' );
+	// This will disable groups feeds.
+	remove_action( 'bp_actions', 'groups_action_group_feed' );
+	// This will disable forums feed.
+	remove_filter( 'bbp_request', 'bbp_request_feed_trap' );
+}
+
+/**
+ * Function will restricate rest API.
+ */
+function bb_restricate_rest_api() {
+	// This will disable woocommerce endpoints.
+	if ( class_exists( 'woocommerce' ) ) {
+		remove_action( 'rest_api_init', array( WC()->api, 'register_rest_routes' ), 10 );
+	}
+	// This will disable all rest api endpoints.
+	add_filter( 'rest_endpoints', 'bb_remove_all_endpoints' );
+}
+
+/**
+ * Function will unset all endpoints.
+ *
+ * @param $endpoints
+ */
+function bb_remove_all_endpoints( $endpoints ) {
+	foreach ( $endpoints as $endpoint => $details ) {
```

Comment:
@jitendrabanjara1991 We need to add a filter here to exclude specific endpoints.
In BuddyBoss APP, If the app is private then we need a specific endpoint to work.

Reply:
@KartikSuthar 
Request changes are done. Please review it.
Thanks.